client/src/actions/projects.ts

```diff
@@ -309,77 +305,93 @@ export const fetchApplicationStatusesFromContract =
             status: projectApplication.status,
           });
         }
-      } catch (error) {
+      } catch (error: any) {
         datadogRum.addError(error, { roundAddress });
         console.error("getApplicationStatusFromContract() error", {
           roundAddress,
           error,
         });
+        dispatch({
+          type: PROJECT_APPLICATIONS_ERROR,
+          projectID,
+          error: error.message,
+        });
       }
     });
   };
 
 export const fetchProjectApplications =
-  (projectID: string, chain: number) => async (dispatch: Dispatch) => {
+  (projectID: string, reactEnv: any /* ProcessEnv */) =>
+  async (dispatch: Dispatch) => {
     dispatch({
       type: PROJECT_APPLICATIONS_LOADING,
       projectID,
     });
 
-    const projectApplicationID = generateUniqueRoundApplicationID(
-      chain,
-      projectID
-    );
+    const { web3Provider } = global;
+
+    await web3Provider?.chains?.forEach(async (chain) => {
+      const projectApplicationID = generateUniqueRoundApplicationID(
+        chain.id,
+        projectID
+      );
```

Comment:
@DanieleSalatti the id for the application is unique per project hashing `chainID,  projectRegistryAddress, projectID` (using the registry address where the project has been created) but with this code we have a different one based on the round chain

Reply:
Good catch!

We make the same mistake when we send the application  (that's why it works). Will fix that as part of this PR.

Reply:
yeah in theory for now everything was on the same network so this starts having problem when we are really multi chain